### ISSUE TITLE
Incompatibility with Windows 8.1

### ISSUE THREAD

**Issue by seemebreakthis** [rdp/on-screen-capture-recorder-to-video-windows-free#3]
This problem seems to exist only with win8.1, and there is a workaround for it.

Repro Steps:
1.  In Win8.1 there is a new option to select window/font sizes that dynamically adjusts to different monitors (useful in case you have more than one monitor attached to your machine).  This option doesn't work well with screen capture recorder.  To repro, first step is to enable this option rather than choosing your system-wide font size the traditional way (100%, 125%, or 150% etc).
2.  Once step one is done, then go to change the monitor to any resolution lower than the native (max) resolution supported by your monitor.
3.  Use the screen capture recorder (via VLC media player for example) and try using the screen capture recorder as the input.

Result:  the screen area occupies only about 3/4 of the entire space so you always get a big area of black stripe at right and bottom.  Also the mouse pointer isn't "in sync" with the captured screen anymore, meaning the pointer cursor seems to still follow the 'original' size, so it can sometimes go 'off screen'.  Sorry it is hard to describe it with words.

Workaround:  if you want to lower the screen resolution before using screen capture recorder, disable that new Windows 8.1 feature first and use the traditional '100%, 125%, 150%' font/window size adjustment option in control panel first.

**Comment by rdp** [comment]
I'll try to test/fix this if I can ever get a windows 8.1 box...

**Comment by rbrandao** [comment]
Hi guys, i'm having some issues using on screen capture on Windows 8.1 (x64) + VLC for recording. I'm getting a scrambled video like this: https://www.dropbox.com/s/iwdt6cnwgvh4ur1/teste.mp4

Preview is working fine, but recording generates this messy video. It also seems to work fine on another machine with Windows 7 x86 + VLC. 

Has anyone saw something like this?

Thanks

**Comment by rdp** [comment]
@rbrandao  what resolution, what version?

**Comment by rbrandao** [comment]
Hi,

Latest version 0.9.7pre, latest VLC 2.1.1 too. Tried to encode with
H264+MP3 (MP4 container) profile.

---

Rafael R.M. Brandão
rafaelbrandao(at)gmail.com

**Comment by rdp** [comment]
what screen resolution?

**Comment by rbrandao** [comment]
1366 x 768

---

Rafael R.M. Brandão
rafaelbrandao(at)gmail.com

**Comment by rdp** [comment]
do you have any DIP settings turned on?

**Comment by rbrandao** [comment]
Sorry, what's DIP?

---

Rafael R.M. Brandão
rafaelbrandao(at)gmail.com

Em 13/12/2013 18:37, "Roger Pack" notifications@github.com escreveu:

> do you have any DIP settings turned on?
> 
> On Fri, Dec 13, 2013 at 12:42 PM, rbrandao notifications@github.com
> wrote:
> 
> > 1366 x 768
> > 
> > ---
> > 
> > Rafael R.M. Brandão
> > rafaelbrandao(at)gmail.com
> >

**Comment by rdp** [comment]
oops I meant DPI
https://github.com/rdp/screen-capture-recorder-to-video-windows-free/issues/31
what does vlc say for 'codec information"

**Comment by rbrandao** [comment]
Codec info is showing an incosistent resolution height:

https://www.dropbox.com/s/b74m0361hkubf3v/Captura%20de%20tela%202013-12-14%2015.11.23.png

Even if I manually specify my resolution in codec settings, it still shows
the overflowed value.

Any hints on how to check/set DPI settings on Windows 8?

**Comment by danvim** [comment]
I have a screen ratio problem, I don't care how big my capture is but my 16:9 desktop on Win8.1 shrank into a 4:3, ant idea?

**Comment by rdp** [comment]
so the recording is 4:3? how is it recorded? wait isn't 4:3 the same as
16:9? :)

**Comment by danvim** [comment]
Oh~ Cheeky Roger 3:) Please try to get it fixed. My screen is 1366x768 16:9, but when I tried to stream a Direct Show using your newest filter on a Windows 8.1, the stream got shrank to a 4:3 with an unknown resolution, so all the graphics and texts are narrow. I tried using the `video size` option below the video and audio device option, but a dialog popped up saying I can't to it for this filter.

 But I want to thank you for your filter, because you gave so much effort into it.
　　　16:9　　　　　　　　　　4:3
－－－－－－－－－ 　 　 －－－－－－－
|　　　／＼　　　 |　　-->|　　　/\ 　　 |<--
|　　　＼／　　 　|　 　　|　　　\/ 　　 |
－－－－－－－－－ 　 　 －－－－－－－

**Comment by rdp** [comment]
are you using vlc to stream? what are you playing it with? how to reproduce?

I've also seen these messages before:
"
Another issue that I have with the screen recorder is that I have a
screen resolution of 1920 x 1080, which is also recorded, but when I
show it in VLC it is in 4:3 display ratio rather than 16:9. I then just
force VLC to use 16:9 but it would be nice if that would work without
workaround :) ."

"C:\Program Files(x86)\VideoLAN\VLC
start vlc -vvv dshow:// --aspect-ratio 16:9
--dshow-vdev="screen-capture-recorder"
--dshow-adev="virtual-audio-capturer" --dshow-aspect-ratio="16:9"
--screen-fps=25.0
:#transcode{sfilter=mosaic,vcodec=mp1v,vb=64,width=1366,height=768,acodec=mp3,ab=32,channels=2,samplerate=44100}:bridge-in{id-offset=100}:display
FWIW

**Comment by danvim** [comment]
Yes, I am using Windows 8.1, the newest VLC and your newest filter.
1. Using the GUI version, Media > Stream
2. Capture Device > CaptureMode: DirectShow; VideoDevice:screen-capture-recorder > Stream
3. Using HTML method and receive locally at http://192.168.x.x:8080

**Comment by rdp** [comment]
I believe it is a VLC bug, does --aspect-ratio 16:9 help?

**Comment by rdp** [comment]
Suspected VLC bug [can avoid it by using the video-aspect parameter, FWIW].
Leaving this open to remind me to ping VLC about it...

**Comment by rbrandao** [comment]
Thank you for the heads up about the video-aspect parameter!

This bug was driving me crazy.

---

Rafael R.M. Brandão
rafaelbrandao(at)gmail.com

**Comment by rbrandao** [comment]
Hi,

Actually, the video-aspect paremeter didn't solve all my problems. It seems
to work with local preview from dshow devices on vlc, but when I stream
through HTTP (TS + h264) the video appears distorted on client. :(

---

Rafael R.M. Brandão
rafaelbrandao(at)gmail.com

**Comment by rdp** [comment]
maybe ping the vlc guys about it?

**Comment by rdp** [comment]
saving the file works ok?

**Comment by rbrandao** [comment]
I figured it out. It's the 1366 x 768 resolution that results in a odd
aspect ratio. if I change resolution in Windows settings to 1360x768 it all
works fine.
